src/efi.rs

```diff
@@ -455,46 +461,84 @@ fn validate_esp(dir: &openat::Dir) -> Result<()> {
     Ok(())
 }
 
-#[context("Clearing current EFI boot entry")]
-pub(crate) fn clear_efi_current() -> Result<()> {
-    const BOOTCURRENT: &str = "BootCurrent";
-    if !crate::efi::is_efi_booted()? {
-        log::debug!("System is not booted via EFI");
-        return Ok(());
+#[context("Get product name")]
+fn get_product_name() -> Result<String> {
+    let content = std::fs::read_to_string(SYSTEM_RELEASE)
+        .with_context(|| format!("Failed to read {SYSTEM_RELEASE}"))?;
+
+    // Define the regex pattern
+    let re = Regex::new(r" *release.*").unwrap();
+
+    // Replace the matched patterns
+    let result = re.replace_all(&content, "").trim().to_owned();
+    log::debug!("Get product name '{result}'");
+    Ok(result)
+}
+
+#[derive(Debug, PartialEq)]
+struct BootEntry {
+    id: String,
+    name: String,
+}
+
+/// Parse boot entries from efibootmgr output
+fn parse_boot_entries(output: &str) -> Vec<BootEntry> {
+    let mut entries = Vec::new();
+
+    for line in output.lines() {
+        // Example line: Boot0002* Fedora	HD(xx)
+        if line.starts_with("Boot0") {
+            if let Some((parts, _)) = line.split_once('\t') {
+                if let Some((id, name)) = parts.split_once(' ') {
+                    let id = id[4..].trim_end_matches('*').to_string();
```

Comment:
Minor but: The `[4..]` part could panic if the input was malformed.  WDYT about this version?

```
/// Parse boot entries from efibootmgr output
fn parse_boot_entries(output: &str) -> Vec<BootEntry> {
    let mut entries = Vec::new();

    for line in output.lines() {
        let line = if let Some(l) = line.strip_prefix("Boot") {
            l
        } else {
            continue;
        };
        let parts = if let Some((parts, _)) = line.split_once('\t') {
            parts
        } else {
            continue;
        };
        if let Some((id, name)) = parts.split_once(' ') {
            let id = id.trim_end_matches('*').to_string();
            let name = name.trim().to_string();
            entries.push(BootEntry { id, name });
        }
    }
    entries
}
```

I was testing this out with https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=638374b60de0116bdf9462c9a8099603

Reply:
Good suggestion, the `filter_map` looks more rust and pretty, thank you very much!

Reply:
To be clear your previous code was great too!  This was a very minor nit.